### PR TITLE
Fix bidirectional search

### PR DESCRIPTION
**Pull request by globalworming**
all possible cases in this environment should work now

### CODE REVIEW

**Comment by globalworming** [comment]
this whole "build a resultpath from two nodes" is pretty ugly. It would be better if there was a way to reverse actions. I guess that's figured out in  #298

File: core/src/main/java/aima/core/search/basic/uninformed/BidirectionalSearch.java [hunk 1]
```diff
@@ -25,86 +25,147 @@
  * The strategy of this search implementation is inspired by the description of
  * the bidirectional search algorithm i.e. Problem is reversed and search is performed from both
  * frontiers.
- * @author manthan
+ * @author wormi
```
Comment (ctjoreilly) [inline core/src/main/java/aima/core/search/basic/uninformed/BidirectionalSearch.java:28]: Please don't replace a prior author. If the work is based on, even slightly, on earlier work all authors should remain listed. You may place your name to the head of the list if the most recent changes are predominantly yours.

**Review by ctjoreilly — APPROVED** [review]
I am going to merge but ask that you address the comment as regards author names.

Thanks

Ciaran

File: core/src/main/java/aima/core/search/basic/uninformed/BidirectionalSearch.java (hunk 1 above)
Comment (globalworming) [reply core/src/main/java/aima/core/search/basic/uninformed/BidirectionalSearch.java:28]: sorry, I will keep that in mind next time

**Pull request merged**